src/lib/flow/sol-flow-node-options.c

```diff
@@ -478,104 +476,176 @@ direction_vector_parse(const struct sol_flow_node_options_member_description *me
-        int i;
-        errno = 0;
-        i = strtol(value, NULL, 0);
-        if ((errno != 0) || (i < 0) || (i > 255)) {
-            SOL_DBG("Invalid byte value for option name=\"%s\": \"%s\"",
-                member->name, value);
-            return -errno;
-        }
-        *byte = i;
-    } else if (streq(t, "int")) {
-        struct sol_irange *i = mem;
-        int r = irange_parse(member, value, i);
-        SOL_INT_CHECK(r, < 0, r);
-    } else if (streq(t, "float")) {
-        struct sol_drange *f = mem;
-        int r = drange_parse(member, value, f);
-        SOL_INT_CHECK(r, < 0, r);
-    } else if (streq(t, "rgb")) {
-        struct sol_rgb *rgb = mem;
-        int r = rgb_parse(member, value, rgb);
-        SOL_INT_CHECK(r, < 0, r);
-    } else if (streq(t, "direction-vector")) {
-        struct sol_direction_vector *direction_vector = mem;
-        int r = direction_vector_parse(member, value, direction_vector);
-        SOL_INT_CHECK(r, < 0, r);
+    if (streq(value, "1") || streq(value, "true") || streq(value, "on") || streq(value, "yes")) {
+        m->boolean = true;
+    } else if (streq(value, "0") || streq(value, "false") || streq(value, "off") || streq(value, "no")) {
+        m->boolean = false;
     } else {
-        SOL_WRN("Unsupported member parse #%u "
-            "name=\"%s\", type=\"%s\", offset=%hu, size=%hu",
-            (unsigned)(member - desc->members), member->name,
-            member->data_type, member->offset, member->size);
+        return -EINVAL;
     }
-
     return 0;
 }
 
-static const struct sol_flow_node_options_member_description *
-find_member(const struct sol_flow_node_options_description *desc, const struct sol_str_slice name)
+static int
+byte_parse(const char *value, struct sol_flow_node_named_options_member *m)
 {
-    const struct sol_flow_node_options_member_description *m;
+    int v;
 
-    for (m = desc->members; m->name != NULL; m++) {
-        if (sol_str_slice_str_eq(name, m->name))
-            return m;
+    errno = 0;
+    v = strtol(value, NULL, 0);
+    if ((errno != 0) || (v < 0) || (v > 255)) {
+        return -EINVAL;
     }
 
-    return NULL;
+    m->byte = v;
+    return 0;
 }
 
 static int
-split_option(const char *input, const char **key, unsigned int *key_len, const char **value)
+string_parse(const char *value, struct sol_flow_node_named_options_member *m)
 {
-    const char *equal = strchr(input, '=');
+    char *s;
 
-    if (!equal || equal == input || equal + 1 == '\0')
-        return -EINVAL;
+    s = strdup(value);
+    if (!s)
+        return -errno;
 
-    *key = input;
-    *key_len = equal - input;
-    *value = equal + 1;
+    m->string = s;
     return 0;
 }
 
+static int(*const options_parse_functions[]) (const char *, struct sol_flow_node_named_options_member *) = {
+    NULL,
+    boolean_parse,
+    byte_parse,
+    irange_parse,
+    drange_parse,
+    rgb_parse,
+    direction_vector_parse,
+    string_parse,
```

Comment:
Might be a good idea to index these by `enum sol_flow_node_options_member_type`:

     [FOO_BOOLEAN] = boolean_parse,
     [FOO_BYTE] = byte_parse,
     [FOO_STRING] = string_parse,